includes/class-sensei-utils.php

```diff
@@ -2626,6 +2626,68 @@ public static function format_last_activity_date( string $date ) {
 
 		return wp_date( get_option( 'date_format' ), $date->getTimestamp(), $timezone );
 	}
+
+	/**
+	 * Render a video embed.
+	 *
+	 * @param string $url The URL for the video embed.
+	 *
+	 * @return string an embeddable HTML string.
+	 */
+	public static function render_video_embed( $url ) {
+		$allowed_html = array(
+			'embed'  => array(),
+			'iframe' => array(
+				'title'           => array(),
+				'width'           => array(),
+				'height'          => array(),
+				'src'             => array(),
+				'frameborder'     => array(),
+				'allowfullscreen' => array(),
+			),
+			'video'  => Sensei_Wp_Kses::get_video_html_tag_allowed_attributes(),
+		);
+
+		if ( 'http' === substr( $url, 0, 4 ) ) {
+			// V2 - make width and height a setting for video embed.
+			$url = wp_oembed_get( esc_url( $url ) );
+			$url = do_shortcode( html_entity_decode( $url ) );
+		}
+		return Sensei_Wp_Kses::maybe_sanitize( $url, $allowed_html );
+	}
+	/**
+	 * Gets the HTML content from the Featured Video for a post
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $post_id the post ID.
+	 *
+	 * @return string|false The featured video HTML output if it exists, or false if it doesn't.
+	 */
+	public static function get_featured_video_html( $post_id ) {
+		$video_embed = '';
+		if ( has_blocks( $post_id ) ) {
+			$post   = get_post( $post_id );
+			$blocks = parse_blocks( $post->post_content );
+			foreach ( $blocks as $block ) {
+				if ( 'sensei-lms/featured-video' === $block['blockName'] ) {
+					// Handle Media Library Files.
+					if ( 'sensei-pro/interactive-video' === $block['innerBlocks'][0]['blockName'] ) {
+						$block = $block['innerBlocks'][0];
```

Comment:
Oh just started testing with the merged branch and there is an issue here, the interactive video block will get lost because we just take the video URL from it. 

I think we can render the whole featured-video block here and return it, just have to make sure that the right content filters are also applied. I'm investigating a bit but so far looks like just running autembed is enough:

```php
$content = render_block( $block );
global $wp_embed;
return $wp_embed->autoembed( $content );
```